### ISSUE TITLE
Installation fails on OS X 10.11 (El Capitan)

### ISSUE THREAD

**Issue by rizkysyazuli** [Benjamin-Dobell/Heimdall#291]
First i got this warning message when i tried to install it.

![screen shot 2015-09-26 at 8 19 33 pm](https://cloud.githubusercontent.com/assets/152221/10117756/2bd613c2-648c-11e5-838e-5cb5a7ba8e73.png)

But i continue anyway, which got me this:

![screen shot 2015-09-26 at 8 24 12 pm](https://cloud.githubusercontent.com/assets/152221/10117771/b90633c6-648c-11e5-9f55-ddf6a5cb03a0.png)

Let me know if you need more info.

**Comment by dgerosa** [comment]
Same thing here

**Comment by yagomont** [comment]
Can't you see, that it is incompatible?

**Comment by thy1985** [comment]
Thing is that heimdall uses an improper location to place its Library and binary during installation:
/usr/bin
and
/usr/lib

When upgrading from a previous version to El Capitan the files are quarantined:
```
/Library/SystemMigration/History/Migration-9BB03294-CD3B-4CF5-9613-5A3B69DBCE14/QuarantineRoot//usr/bin/heimdall
/Library/SystemMigration/History/Migration-9BB03294-CD3B-4CF5-9613-5A3B69DBCE14/QuarantineRoot//usr/lib/libusb-1.0.dylib
```

As they in fact should not lurk around there. A more proper location is /usr/local/ (or older /opt/local) prefix. This is what Apple also recommends.

I didn't check it so far but my guts feeling is that the installer is probably trying to write in those directories which is not possible anymore (straight forward) even with elevated privileges.

That means  you probably need to wait for the developer to make the installer point to a more suitable location or you copy the files from quarantine to somewhere else (for cli only use) [NOTE: The UUID in the path is for sure a different on your systems ;)]

**Comment by Benjamin-Dobell** [comment]
> As they in fact should not lurk around there. A more proper location is /usr/local/ (or older /opt/local) prefix. This is what Apple also recommends.

It might be what Apple recommends *now*, but it's certainly not how Unix filesystems have ever worked in the past. ```/usr/bin``` and ```/usr/lib``` are with out a doubt the correct place for this kind of software. If Apple are stating otherwise, they're blatantly wrong. Do you have a reference?

```/usr/local``` is for *locally* built software i.e. software you've compiled yourself. So when you make Heimdall from source code it should go in ```/usr/local```. ```/bin``` and ```/lib``` on the other hand are system directories, these are reserved for use by the OS maintainer, in this case Apple. ```/usr/*``` on the other hand is for user installed, but third-party vendor built/maintained software.

The reason Heimdall doesn't work on El Capitan is because Apple require the binaries, kexts and installer to be signed. I (via my company Glass Echidna) have the necessary signing certificates to sign all of the above (including the kext). It's just that the last officially packaged releases of Heimdall were literally made years ago, well before El Capitan was around,

When I finally get a chance I would love to make a new release - there's just a lot of poorly supported devices at the moment, so I don't feel comfortable doing so. The reason the devices are poorly supported is because I don't have access to the vast majority of Samsung's more recent devices (they release far too many variants each year).

Anyway, when I get a chance, hopefully soon, I'll release an installer for kext only - because that's something most people won't be able to build and run on their on own system.

**Comment by thy1985** [comment]
Hi Benjamin

Yay, sorry. I did not mean to offend you/anyone. Sorry if this sounded a bit hash!! Your work on heimdall is really much appreciated!
And it works fine on the El C when you cp it to a different location (of whatever name and purpose).
A signed installer that can put the stuff in place is surely a great thing.

Still, Apple's OS is no longer a system that works how "unix has ever worked before"...
I never said that i agree to the path Apple took...
You are right with the semantics for locally built software. :) Nevertheless the location can be used at least for a workaround period.
Cheers,

**Comment by flybd5** [comment]
Ouch. Until this is fixed I'm stuck with a Note 10.1 that needs reflashing. :(

**Comment by Royedc4** [comment]
thy1985 What files do i have to cp? I didn't have any previous versions... Thanks

**Comment by Benjamin-Dobell** [comment]
I'm not running El Capitan, so this is 100% untested, but please give this a shot:

https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg

It's a signed copy of Heimdall CLI (and the corresponding kext), version "whatever was sitting in my source directory" :stuck_out_tongue:

**Comment by Royedc4** [comment]
Successfully installed on El Capitan ;)

**Comment by Benjamin-Dobell** [comment]
Great to hear.

**Comment by Benjamin-Dobell** [comment]
Until I get around to making an official release, if you want a version other than "whatever was sitting in my source directory", please install the package above (as you'll need the signed kext installed by a signed installed). Then you can go and build any other version of Heimdall from source.

**Comment by Roman2K** [comment]
Sorry, I tried to install `Heimdall-1.4.1-Unofficial-Signed.dmg` in El Capitan and it failed:

<img width="732" alt="screen shot 2015-10-09 at 16 42 53" src="https://cloud.githubusercontent.com/assets/2480/10397137/6d6cabc4-6ea6-11e5-8ddf-146234dfb1bd.png">
And the only relevant error in system.log:

```
Oct  9 16:52:30 Romans-iMac sandboxd[128] ([874]): mkdir(874) System Policy: deny file-write-create /usr/local
```

I tried to create `/usr/local` by hand but it won't either:

```shell
~$ sudo mkdir /usr/local
Password:
mkdir: /usr/local: Operation not permitted
```

Which to me is a good thing but I need Heimdall :confused:

**Comment by Benjamin-Dobell** [comment]
Does the "Allow apps downloaded from:" setting still exist?

![screen shot 2015-10-10 at 2 05 59 am](https://cloud.githubusercontent.com/assets/482276/10397447/828a06b2-6ef3-11e5-807c-0d1d1f564cc5.png)

If so, make sure you've enabled "and identified developers".

Everything I've read and seen (WWDC talks) explicitly state that ```/usr/local``` is still the correct location for third-party (non app store) software. ```/usr/local``` is specifically an exception in ```/System/Library/Sandbox/rootless.conf``` which says that SIP is disabled. The gory details can be found on stack overflow:

http://apple.stackexchange.com/questions/193368/what-is-the-rootless-feature-in-el-capitan-really

**Comment by Roman2K** [comment]
Thanks for the details about `/usr/local`. And the setting still exists:
<img width="780" alt="screen shot 2015-10-09 at 17 10 41" src="https://cloud.githubusercontent.com/assets/2480/10397588/cb764e12-6ea8-11e5-87cd-88baab63e445.png">

**Comment by Benjamin-Dobell** [comment]
Sp as far as I can tell Apple are publicly like:

> No way! We're not preventing users from installing third-party software. You can still write to "/usr/local"

_..in private_:

> ... if it existed. Muahahaha!

As far as I can tell, fresh installs of El Capitan have an exception to allow software to be installed and run in ```/usr/local```, it's just that the directory doesn't exist... and a user can't even create it! Not even root.

The only way to create it is by pulling messing around with the system in recovery mode e.g.

http://digitizor.com/2015/10/01/fix-homebrew-permissions-osx-el-capitan/

**Comment by sunny-mittal** [comment]
I'm running El Capitan and have a /usr/local directory...I'm not sure is the issue people are having but your unofficial installer worked fine for me. Thanks!

**Comment by traveller1** [comment]
Thanks, 1.4.1 worked for me. mba 10.11.1

**Comment by coleHafner** [comment]
The unofficial Heimdall-1.4.1-Unofficial-Signed.dmg worked for me! Thank you! 

https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg

**Comment by sleepingscorpion** [comment]
The 'Heimdall-1.4.1-Unofficial-Signed.dmg' worked fine for me too on El C. Cheers!

**Comment by kevinhooke** [comment]
Heimdall-1.4.1-Unofficial-Signed.dmg worked for me too on OS X 10.11.1

**Comment by thalesmello** [comment]
+1: maybe put this on the home page of the project? Was a little bit hard to find this thread.

**Comment by gslender** [comment]
+1: yep, needed this unofficial signed release too for me on OS X 10.11.1

**Comment by kalenjohnson** [comment]
Just found and successfully installed 1.4.1 :+1: 

Don't know why someone wouldn't have `/usr/local`, seems strange.

**Comment by matgessel** [comment]
@Benjamin-Dobell Perhaps this issue should stay open until 1.4.1 is posted on glassechidna. I suspect that people are creating duplicate issues because this issue is filtered out of the default view.

**Comment by lboix** [comment]
Thank you very much for providing to us the unofficial Heimdall-1.4.1-Unofficial-Signed.dmg 

https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg

Same situation here, and all worked fine with this package!

**Comment by simsketch** [comment]
thanks for the link Iboix!

**Comment by GoncaloPT** [comment]
Thank you  for providing 1.4.1. 
You have my positive feedback runing el capitan

**Comment by kivikakk** [comment]
+1 for working on 10.11. I don't think SIP is responsible for the failure here.

**Comment by SephVelut** [comment]
+1

**Comment by hoanghua** [comment]
+1 works for me with last El Capitan (10.11.3). Thanks

**Comment by hiddentao** [comment]
+1 unofficial installer works for me too (El Capitan 10.11.2). Thanks!

**Comment by nikhilninawe** [comment]
It installed for successfully for me too. Thanks!!

**Comment by mkraft** [comment]
@Benjamin-Dobell Thanks very much for your build.

**Comment by Gustto** [comment]
Can anyone explain me how I install 1.4.1 and use it with the frontend/suite?
I'm having trouble connecting to my Samsung S4 using Heimdall on OSX (10.10.5) and hope 1.4.1 will help me. 
Thanks all!

**Comment by rahulvenus** [comment]
Thank you for your build ....worked for me too...

**Comment by grizmt** [comment]
Installation says successful, however the front-end is not in my applications folder. Running 10.11.3, any insight?

**Comment by Benjamin-Dobell** [comment]
@grizmt Sorry, it's just the command line tool in that package.

**Comment by grizmt** [comment]
@Benjamin-Dobell Oh right, I didn't realise. Thanks!

**Comment by QueenAdrienne** [comment]
Thanks @Benjamin-Dobell appreciate the information and the working file. It's clear that Apple is moving toward restricting their environment to the point that unless Apple can monetize it, it's not happening. The Apple/iphone environment is being insulated.I sent my iphone 6 back. Can't download ring tones, Can't install non itunes apps,etc. I think my next laptop will be either Android, Unix, or god forbid Windows.

**Comment by FrustratedTechNoob** [comment]
Thanks for the solution - it installed for me fine (1st time ever i have had an issue, googled it and the first result solved it). Now to the next step of getting my s3 running cyanogen...

**Comment by jeridgas** [comment]
Can anyone help? It says the install was successful yet I can't locate 1.4.1 anywhere......any thoughts on this?

**Comment by jeridgas** [comment]
@Benjamin-Dobell where do I get the proper frontend package for this version?

**Comment by ronparchita** [comment]
I have Mavericks and I have the same problem
At first I downloaded from http://mac.softpedia.com/get/Utilities/Heimdall.shtml v1.4.1
2 clicks, reboot, nothing in Applications at heimdall.
So I see here in this tread another link. I downloaded it, 2 clicks, reboot.
It is the same. Nothing, nada, rien in the applications folder !
Why, I don't understand what happens.
Please, can you help me ?
Please I'd like to know if it's ok with this download only or if I have to download others folders.

Many thanks

R

**Comment by jonn26** [comment]
If you want a front-end for the heimdall cli try JOdin3.

https://jodin3.casual-dev.com/

**Comment by ronparchita** [comment]
Thank you jonn26

Please, can you tell mi what I have to do with that :+1: 
![capture d ecran 2016-04-05 a 15 26 46](https://cloud.githubusercontent.com/assets/18006491/14283381/7e3e5248-fb43-11e5-9a2c-437d286703e7.jpg)

Thank you

**Comment by thelangosta** [comment]
The Jodin3 site has been down since yesterday. I'm not sure if Adam Outler decided to stop the server since he doesn't seem to support Jodin any more. It's getting harder to find the right tools to restore a bricked samsung. Fastboot would make things much easier.

**Comment by StevenXL** [comment]
Installed https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg on OS X El Capitan (10.11.4). Worked! Thank you.

**Comment by KingHebes** [comment]
This does not work on my version of 10.11.4. That download file is much smaller than the official version from http://glassechidna.com.au/heimdall/ I installed the file that @Benjamin-Dobell put up top in his post and then tried the official version and it still did not install.

**Comment by mrgame64** [comment]
The Unsigned installer works on OSX 10.11 here too! Thanks again :)

**Comment by weareshipco** [comment]
Installed perfectly on OSX 10.11 here as well. Thank you!

**Comment by fawyna** [comment]
Installed perfectly on OSX 10.11 here as well. Thank you!

**Comment by gcheron** [comment]
Hi,

" Failed to confirm end of file transfer sequence! ERROR: RECOVERY upload failed! "

I managed to install the unofficial package under El Capitan (thank you very much!) but when it is used with JOdin3 I'm getting an issue when I try to root my Note 4. Heimdall managed to automatically download the PIT with JOdin but fails when using the flash command. I run JOdin3 from a terminal and here is the output:

[...]
[DEBUG]Run Heimdall from DeviceCommunicationProtocol:/usr/local/bin/heimdall flash --PIT /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/CASUALroot-2016-06-30-10.37.13/part.pit --CACHE /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/CASUALroot-2016-06-30-10.37.13/cache.img.ext4 --RECOVERY /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/CASUALroot-2016-06-30-10.37.13/recovery.img
[DEBUG]###executing real-time command: /usr/local/bin/heimdall###
[ERROR]ANALYSIS DETECTED ERROR:Failed to confirm end of file transfer sequence!
[DEBUG]State Change Detected, The new state is: 0
[DEBUG]@stateDisconnected
[DEBUG]Device Removed
[DEBUG]Run Heimdall from DeviceCommunicationProtocol:/usr/local/bin/heimdall detect
[VERBOSE]Heimdall Device detected!
[VERBOSE]Starting: mode:Heimdall
Connected:true

[DEBUG]Run Heimdall from DeviceCommunicationProtocol:/usr/local/bin/heimdall detect
[VERBOSE]Heimdall Device detected!
[DEBUG]State Change Detected, The new state is: 1
[DEBUG]Device Connected
[DEBUG]@stateConnected

Any idea?

**Comment by BirchardEricAvery** [comment]
That was awesome, I've never gotten this to work--until now.  

Running OSX 10.11.5 and all I did was blindly copy/ paste to every location mentioned--it worked the very first time.  So the two quarantined files are correct and run perfectly.  

Should on choose to, you can always delete the extra copies, but it works so not going to fix something that isn't broke.  However, if some else reads this try one place at time so as to find the most correct spot.  It's only two files and a few locations which leaves on a few combinations.

**Comment by ttimz1313** [comment]
Cool

**Comment by luizfelipetx** [comment]
Works for me, 

@ttimz1313  update the link in website. From there is not working.
http://glassechidna.com.au/heimdall/#downloads (1.4.0 no work in el c, 1.4.1 unsg works fine.)

**Comment by cederom** [comment]
> > As they in fact should not lurk around there. A more proper location is /usr/local/ (or older /opt/local) prefix. This is what Apple also recommends.
> 
> It might be what Apple recommends now, but it's certainly not how Unix filesystems have ever worked in the past. /usr/bin and /usr/lib are with out a doubt the correct place for this kind of software. If Apple are stating otherwise, they're blatantly wrong. Do you have a reference?

Apple OSX/macOS is a BSD Unix OS. You can easily see exactly the same approach in FreeBSD of separation /bin and /usr/bin as the OS BASE from /usr/local USERLAND. This is because base may be booted from another place than userland. Also noone wants to modify base. :-)

**Comment by matiasnj** [comment]
Successfully installed also on mac OS Sierra

**Comment by tutejszy** [comment]
Confirm successful install 1.4.1 on MacOS 10.12.2 Sierra

**Comment by papei** [comment]
I installed the unofficial Heimdall-1.4.1-Unofficial-Signed.dmg with success. However after installing it I went to update my Homebrew and got the following warning.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
/usr/local/lib/libusb-1.0.dylib

Should I delete libusb-1.0.dylib or leave it there of the time being and see what happens? If I delete, will it have an effect on Heimdall?

Running OSX 10.11.6

**Comment by psrrao** [comment]
I'm running macOS 10.12.1, installed Heimdall-1.4.1-Unofficial-Signed.dmg successfully.

Thanks..!!

**Comment by jeff303** [comment]
For the folks saying the Heimdall-1.4.1-Unofficial-Signed.dmg installer worked, where do you find the Heimdall Frontend after installing?  I've tried Finder and the locate tool from the command line and am unable to locate any files after it completes.  As someone else noted, it's around 120kb whereas the 1.4.0 dmg file (which doesn't work on Sierra) is 7.3MB, so something is clearly off there.

**Comment by papei** [comment]
I have it right in my applications. I've never used the frontend though, so I can't tell you if it is working properly or not. I use heimdall exclusively from the CLI. 

![screenshot 2017-01-22 19 22 58](https://cloud.githubusercontent.com/assets/11274073/22184722/63cbd432-e0d8-11e6-96fb-623dce2f1180.png)

**Comment by JinQzzz** [comment]
`Heimdall-1.4.1-Unofficial-Signed.dmg` working on `Sierra 10.12.3`

**Comment by giacomodeglinnocenti** [comment]
The unofficial release worked for me too on El Capitan 10.11.6

**Comment by NewbiefromNJ** [comment]
I am currently running OS X 10.11.6 El Capitan. I was not able to install Heimdall-1.4.1-Unofficial-Signed unless "Allow apps downloaded from: Anywhere" was turn on in System Preferences/Security & Privacy. Once I changed this setting, installation was successful. 

However, I cannot find heimdall-frontend anywhere. I've rebooted system, tried installing it again and heimdall-frontend is still missing from Applications folder. Anyone have any recommendations?

**Comment by NewbiefromNJ** [comment]
Just read all comments and learned there is no frontend in the unofficial-signed package - only command line. Is there no way to run Heimdall-frontend on El Capitan?

**Comment by DmitTrix** [comment]
(I don't see this clearly stated in the thread, so let me post it here)
Looks like one still needs to have 1.4.0 version to be installed to have the GUI frontend. As mentioned in Benjamin's comment from 8 Oct 2015, the 1.4.1 package here only contains the kext (kernel extension) and the command-line tool, not the GUI.

**Comment by majuss** [comment]
Its due to "rootless" system of macOS. You basically can't write anymore into /usr. Please use /opt (like macports) instead... This would solve all problems with juste one line :(

**Comment by DanielAlcon** [comment]
unnoficial signed worked for me on Sierra 10.12.5

**Comment by joaoyc** [comment]
I was facing the same issue that most people had (when trying to install Heimdall 1.4.0 in Sierra) in this thread, but I think I could come up with something - based on majuss suggestions -.

since the installation was failing because it tried to install heimdall in /usr/ path, I thought about repackaging the .pkg, but this time referencing the binaries to /usr/local/ instead.

1. Download heimdall-suite-1.4.0-mac.dmg and extract the pkg file within the container into your preferred destination;
2. Extract the pkg's contents into a temp folder by typing the following in a terminal:
> $ mkdir temp; cd temp; xar -xf ../Heimdall\ Suite\ 1.4.0.pkg 
> $ ls -la
total 14520
drwxr-xr-x   7 joao  staff      238 Jun 20 23:42 .
drwx------+ 42 joao  staff     1428 Jun 20 23:42 ..
-rw-r--r--   1 joao  staff   165329 Jun 12  2013   Bom
-rw-r--r--@  1 joao  staff      628 Jun 12  2013    PackageInfo
-rw-r--r--   1 joao  staff  7256659 Jun 12  2013  Payload
-rw-r--r--   1 joao  staff       80 Jun 12  2013       Scripts
3. Inside the tmp folder, you will see a file called Payload - which is an archive of the actual installables. Extract it by typing the following in a terminal:
> $ cat Payload | gunzip -dc |cpio -i
> $ ls -l
total 14520
drwxrwxr-x  3 joao  staff      102 Jun 20 23:50 Applications
-rw-r--r--  1 joao  staff   165329 Jun 12  2013 Bom
-rw-r--r--@ 1 joao  staff      628 Jun 12  2013 PackageInfo
-rw-r--r--  1 joao  staff  7256659 Jun 12  2013 Payload
-rw-r--r--  1 joao  staff       80 Jun 12  2013 Scripts
drwxr-xr-x  3 joao  staff      102 Jun 20 23:50 System
drwxr-xr-x  2 joao  staff       68 Jun 20 23:42 temp
drwxr-xr-x  4 joao  staff      136 Jun 20 23:50 usr
4. Create a new dir called /usr/local/ and move the bin and lib folders to it:
> $ mkdir usr/local; mv usr/bin usr/local; mv usr/lib usr/local
5. Remove the old Payload archive (you will recreate it)
> $ rm -f Payload
6. repackage the .pkg file by putting back all together from our temp dir:
> $ pkgutil --flatten temp/ heimdall.pkg
$ ls -l heimdall.pkg
-rw-r--r--  1 joao  staff  22688524 Jun 20 23:18 heimdall.pkg
7. double click the pkg file and proceed to reinstall it. 


So basically that is it. After doing all these steps, I was able to install `heimdall 1.4.0` in `Sierra 10.12.3` just fine.

Before the fix:
![image](https://user-images.githubusercontent.com/2332093/27365484-b1c3434c-5616-11e7-8a1d-7fd4e0492973.png)


After the fix:
![image](https://user-images.githubusercontent.com/2332093/27365411-639f69e8-5616-11e7-9d49-75143543fb2e.png)

Thanks.

**Comment by attacus** [comment]
@Benjamin-Dobell - thank you very much for your work on Heimdall. If possible, could you upload a new signed dmg? The certificate for your original upload has expired, and the "installation failed" issue persists in Sierra.

**Comment by Lucasjh** [comment]
I'm trying to root my phone and I got the same problem in Sierra. I found this and it worked!: https://forum.xda-developers.com/tab-s2/general/guide-how-to-jodin3-osx-el-capitan-t3393606

**Comment by PauloZumach** [comment]
And works for OSx Sierra as well .... Yessssss!!!!

**Comment by pascalwhoop** [comment]
So I did as @joaoyc suggested but my /usr/local/lib and /usr/local/bin folders were still not holding the files they were supposed to. So I just manually moved all the files to the appropriate places. Kext had also be manually loaded using
`sudo kextload /System/Library/Extensions/heimdall.kext/`
but still it gives me the errors as reported here
https://github.com/Benjamin-Dobell/Heimdall/issues/137

still investigating...

**Comment by yoshiserry** [comment]
I have a somewhat old now but still fresh installation of osx sierra and I found this unofficial build installed correctly. The application in Finder is then called heimdall-frontend. [Unofficial 14.1 build on bitbucket](https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg)

**Comment by pedrosanta** [comment]
Installing the build mentioned in the issue description, worked for me.

**Comment by bayfeynman** [comment]
hey,
i have problem! 
when I type code which is "mkdir temp; cd temp; xar -xf ../Heimdall\ Suite\ 1.4.0.pkg" and then I got message : 
mkdir: temp: File exists
Error opening xar archive: ../Heimdall Suite 1.4.0.pkg
why I got error? any solution of this? I try to find problem but I can't.. if you know solution or help to me, I will be grateful to you..

**Comment by Tenalach** [comment]
can confirm this works for OSX sierra 10.12.6 
First installed Heimdall Suite 1.4.0 which ran an error saying not compatible. Then installed Heimdall-1.4.1-Unofficial-Signed.dmg (as found above) and it worked fine.

**Comment by cannedbrain** [comment]
I successfully installed Heimdall-1.4.1-Unofficial-Signed.dmg but when trying to create PIT file it returned error and only created a corrupted file.
`Initialising connection...
Detecting device...
Claiming interface...
Setting up interface...
ERROR: Setting up interface failed!
Releasing device interface...`

My post on another board: [https://forum.xda-developers.com/showpost.php?p=74575479&postcount=418](url)

**Comment by killuadk** [comment]
hello i bypass it by installing it at different partition

**Comment by sfaul** [comment]
Unofficial v1.4.1 installs successfully in macOS 10.13.1 (17B48)

**Comment by albag57** [comment]
Hello sfaul,
could you explain which procedure did you follow in installing v1.4.1? that because me too I installed succesfully but I do not find where the app has been installed. It seems to be unnamed, so.... Thanks in advance.

**Comment by sfaul** [comment]
@albag57 
It's a commandline tool. To check if it is correctlly installed open Terminal and type: `heimdall`.
If it is correctlly installed, you should get the usage description of heimdall in return.
If not, try a reinstall.

**Comment by albag57** [comment]
Thank you very much, I'll try.
Ciso

Alberto
tel. 3358210107

Il 02 Feb 2018 22:49, "sfaul" <notifications@github.com> ha scritto:

@albag57 <https://github.com/albag57>
It's a commandline tool. To check if it is correctlly installed open
Terminal and type: heimdall.
If it is correctlly installed, you should get the usage description of
heimdall in return.
If not, try a reinstall.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Benjamin-Dobell/Heimdall/issues/291#issuecomment-362717828>,
or mute the thread

**Comment by albag57** [comment]
Sfaul, thank you so much, it worked.

**Comment by anstimp** [comment]
The unofficial Heimdall-1.4.1-Unofficial-Signed.dmg worked for me! Thank you!

https://bitbucket.org/benjamin_dobell/heimdall/downloads/Heimdall-1.4.1-Unofficial-Signed.dmg
